core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.ts

```diff
@@ -134,33 +140,35 @@ export class ExplorationFooterComponent {
       windowClass: 'oppia-modal-lesson-information-card'
     });
 
-    let displayedCardIndex = this.playerPositionService.getDisplayedCardIndex();
-
     modalRef.componentInstance.checkpointCount = this.checkpointCount;
-    // Note to developers:
-    // The checkpointArray is used to track the number of
-    // completedCheckpoints. For the first card, 1 is pushed
-    // since the first card is always a checkpoint. For
-    // displayedCardindex > 1, 1 is pushed if the card is a
-    // checkpoint, else 0 is pushed.
-    let completedCheckpoints = 0;
-    for (let i = 0; i <= displayedCardIndex; i++) {
-      completedCheckpoints = completedCheckpoints + this.checkpointArray[i];
+
+    this.roebas.loadLatestExplorationAsync(this.explorationId).then(
+      response => {
+        if (response.most_recently_reached_checkpoint_state_name) {
+          this.mostRecentlyReachedCheckpointStateName = (
+            response.most_recently_reached_checkpoint_state_name);
```

Comment:
Actually, since the field is populated at the initialization of the exploration itself, it would never encounter None. But, removed the check since it's not really required.